src/java/org/apache/cassandra/cache/ChunkCache.java

```diff
@@ -97,52 +94,109 @@ public class ChunkCache
                                               ? new ChunkCache(BufferPools.forChunkCache(), DatabaseDescriptor.getFileCacheSizeInMB(), ChunkCacheMetrics::create)
                                               : null;
 
+    @Unmetered
     private final BufferPool bufferPool;
 
-    // Relies on the implementation detail that keys are interned strings and can be compared by reference.
-    // Because compute optimistically updates the set without acquiring a lock, we must use a set that is
-    // safe for concurrent access.
-    private final NonBlockingIdentityHashMap<String, NonBlockingHashSet<Key>> keysByFile;
-    private final AsyncCache<Key, Buffer> cache;
-    private final Cache<Key, Buffer> synchronousCache;
-    private final ConcurrentMap<Key, CompletableFuture<Buffer>> cacheAsMap;
+    private final AsyncCache<Key, Chunk> cache;
+    private final Cache<Key, Chunk> synchronousCache;
+    private final ConcurrentMap<Key, CompletableFuture<Chunk>> cacheAsMap;
     private final long cacheSize;
+    @Unmetered
     public final ChunkCacheMetrics metrics;
+    @Unmetered
     private final ShutdownableExecutor cleanupExecutor;
 
     private boolean enabled;
     private Function<ChunkReader, RebuffererFactory> wrapper = this::wrap;
 
+    // Global file id management
+    private static final ConcurrentHashMap<File, Long> fileIdMap = new ConcurrentHashMap<>();
```

Comment:
why this is static and not per-instance ?

even if we are not currently using it, we have 2 instances of the ChunkCache.

check CNDB IndexChunkCache
the idea is to have a dedicaded ChunkCache for SAI

currently I have never seen that it helps, so we could drop it

by the way, the two cache instances are not expected to cache the same files, the IndexChunkCache caches only SAI components

Reply:
Done, it does not need to be static.